eng/Versions.props

```diff
@@ -236,7 +236,7 @@
     <CastleCorePackageVersion>4.2.1</CastleCorePackageVersion>
     <FSharpCorePackageVersion>4.2.1</FSharpCorePackageVersion>
     <GoogleProtobufPackageVersion>3.8.0</GoogleProtobufPackageVersion>
-    <GrpcAspNetCorePackageVersion>2.23.2</GrpcAspNetCorePackageVersion>
+    <GrpcAspNetCorePackageVersion>2.24.0</GrpcAspNetCorePackageVersion>
```

Comment:
@JamesNK what helps this change meet the 3.1 bar? @Pilchie any objections?

Reply:
More specifically, how do the fixes in https://github.com/grpc/grpc-dotnet/releases/tag/v2.24.0-pre1 affect the gRPC features of this repo?

Reply:
Updated customer impact.